eng/scripts/Invoke-DependencyCheck.ps1

```diff
@@ -0,0 +1,77 @@
+Param(
+    [string] $PackageDirectory
+)
+
+. (Join-Path $PSScriptRoot .. common scripts common.ps1)
+
+$sdks = Get-AllPackageInfoFromRepo
+
+## Create depcheck module
+$workingPath = Join-Path $RepoRoot "sdk" "depcheck"
+if (Test-Path -Path $workingPath)
+{
+    Remove-Item -Path $workingPath -Recurse -Force
+}
+New-Item -ItemType Directory -Force -Path $workingPath
+
+## Init go mod
+Set-Location $workingPath
+Write-Host "##[command]Executing go mod init in " $workingPath
+go mod init github.com/Azure/azure-sdk-for-go/sdk/depcheck
+if ($LASTEXITCODE) { exit $LASTEXITCODE }
+
+# Find whether latest version is in the `retract` section in `go.mod` to judge whether the package is temporarily deprecated. 
+function IsPackageDeprecated($sdk)
+{
+    $RETRACT_SECTION_REGEX = "retract\s*((?<retract>(.|\s)*))"
+    $modContent = Get-Content (Join-Path $sdk.DirectoryPath 'go.mod') -Raw
+    if ($modContent -match $RETRACT_SECTION_REGEX) {
+        return $($matches["retract"]).Indexof($sdk.Version) -ne -1
+    }
+}
```

Comment:
Should we explicitly return false here? So we don't get some strange null behavior?

Reply:
Fixed.